src/analyze/analyze.c

```diff
@@ -472,6 +472,7 @@ static int acquire_host_info(sd_bus *bus, struct host_info **hi) {
         return 0;
 fail:
         free_host_info(host);
+        *hi = NULL;
```

Comment:
Why? We have to rule to never touch such return variables unless we return 0. Please make the caller aware of such conditions.